### ISSUE TITLE
Python Expressions

### ISSUE THREAD

**Issue by sfmskywalker** [elsa-workflows/elsa-core#4164]
In addition to having C# + JavaScript + Liquid expressions, we should add Python as an option as well, which is useful when handling data.

**Comment by Tealons** [comment]
I was looking at Elsa for my custom monitoring application. We now use a lot of python scripts for the monitoring jobs, so I was happy to see that Elsa 3.0 has Python planned. I think this is not yet available on the v3 branch? 

And maybe it would be an idea to have both IronPython and Python.NET supported? Because IronPython does not work with some popular packages like Pandas.

**Comment by Tealons** [comment]
Found the module. Going to play around with it and see if I can add a Python.NET support.

**Comment by sfmskywalker** [comment]
> And maybe it would be an idea to have both IronPython and Python.NET supported?

We should definitely add support for Python.NET. Do you think it makes sense to have both IronPython and Python.NET, or would it be enough to just replace the IronPython implementation with Python.NET?

@mohdali Any thoughts on that?

**Comment by mohdali** [comment]
> > And maybe it would be an idea to have both IronPython and Python.NET supported?
> 
> We should definitely add support for Python.NET. Do you think it makes sense to have both IronPython and Python.NET, or would it be enough to just replace the IronPython implementation with Python.NET?
> 
> @mohdali Any thoughts on that?

I think Python.NET is more suitable for most of the use cases that require python integration as it supports more popular Python libraries and might be more performant. However, it requires more setup to install Python externally. IronPython is easier to deploy but a bit limited. We definitely need to add Python.NET support and I think it will be the most common case. However, we need to make sure it is easy to wire it up in all deployment scenarios (windows, linux, docker, etc.)

**Comment by sfmskywalker** [comment]
Thanks for that input. I agree, and so I think that we only need an Python.NEt implementation because it covers all intended use cases - having an IronPython implementation might not add any value, unless there is some use case that requires direct access to .NET without interop between Python.NET and .NET. So far, I have successfully managed to make .NET calls from Python.NET and effectively replaced IronPython. I also tested e.g. `import random` to see if that worked, which it did.
